include/hook.h

```diff
@@ -109,6 +107,15 @@ typedef struct
 	const char *error;
 } hook_data_channel_approval;
 
+typedef struct
+{
+	struct Client *client;
```

Comment:
Maybe include the target explicitly in the struct? So it doesn't have to be fetched from the membership.

```suggestion
	struct Client *client;
	struct Client *target;
```